controller/uninstall_controller.go

```diff
@@ -541,6 +591,43 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
 	return
 }
 
+func (c *UninstallController) deleteShareManagers(shareManagers map[string]*longhorn.ShareManager) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "Failed to delete share managers")
+	}()
+	for _, sm := range shareManagers {
+		log := getLoggerForShareManager(c.logger, sm)
+
+		timeout := metav1.NewTime(time.Now().Add(-gracePeriod))
+		if sm.DeletionTimestamp == nil {
+			if err = c.ds.DeleteShareManager(sm.Name); err != nil {
+				err = errors.Wrapf(err, "Failed to mark for deletion")
+				return
+			}
+			log.Info("Marked for deletion")
+		} else if sm.DeletionTimestamp.Before(&timeout) {
+			var pod *v1.Pod
+			pod, err = c.ds.GetPod(sm.Name)
```

Comment:
I don't think the pod name is `sm.Name`, can you check again and verify if the share managed pods actually deleted by creating a RWX volume.

https://github.com/longhorn/longhorn-manager/blob/master/controller/share_manager_controller.go#L703-L703

Reply:
One can use the function `types.GetShareManagerPodNameFromShareManager` to map a share-manager name to a pod name. See https://github.com/longhorn/longhorn-manager/blob/3b83f478cdcaf70176127e73510d91ae37a9301c/types/types.go#L355-L361

Reply:
changed